src/controllers/StaffController.js

```diff
@@ -553,6 +625,580 @@ class StaffController {
+   */
+  static async adminCreateNew(req, res) {
+    // extract the content of the request
+    const data = await authClient.staffAttrCheck(req);
+    if (data.error) {
+      return res.status(400).json({ error: data.error });
+    }
+    const adminFxn = await authClient.staffPreCheck(req);
+    if (adminFxn.error) {
+      return res.status(401).json({
+        error: adminFxn.error,
+        msg: 'Staff is not authorized to perform this action',
+      });
+    }
+    const { staff, token } = adminFxn;
+    const typeCollection = data.collection;
+    if (!typeCollection) {
+      return res.status(400).json({
+        error: 'Missing collection type for the request',
+        msg: 'Ensure the key is defined as \'collection\'',
+      });
+    }
+    let newObjData;
+    try {
+      newObjData = await staff.createNew(typeCollection, data);
+      if (newObjData.error) {
+        return res.status(400).json({
+          error: newObjData.error,
+        });
+      }
+    } catch (err) {
+      return res.status(400).json({ error: err });
+    }
+    if (!newObjData) {
+      return res.status(400).json({ error: 'Internal Server Error', msg: 'Failed to create new object' });
+    }
+    const operatorData = await staff.getDashboardData();
+    if (!operatorData) {
+      return res.status(400).json({ error: 'Internal Server Error fetching Dashboard' });
+    }
+    return res.status(200).json({
+      operation: 'Object created successfully',
+      xToken: token,
+      operatorData,
+      newObjData,
+    });
+  }
+
+  /**
+   * Updates an object in the specified collection for admin access.
+   * @async
+   * @static
+   * @param {Object} req - The request object.
+   * @param {Object} res - The response object.
+   * @returns {Promise<Object>} The result of updating an object.
+   * @throws {Error} If there is an error during the update process.
+   */
+  static async adminUpdate(req, res) {
+    // Extract the ID from the request parameters
+    const { id } = req.params;
+    if (!id) {
+      return res.status(400).json({ error: 'Missing ID' });
+    }
+    // extract the content of the request
+    const data = await authClient.staffAttrCheck(req);
+    if (data.error) {
+      return res.status(400).json({ error: data.error });
+    }
+    const adminFxn = await authClient.staffPreCheck(req);
+    if (adminFxn.error) {
+      return res.status(401).json({
+        error: adminFxn.error,
+        msg: 'Staff is not authorized to perform this action',
+      });
+    }
+    const { staff, token } = adminFxn;
+    // ascertain if the tokenToObj matches the given ID
+    const typeCollection = data.collection;
+    if (!typeCollection) {
+      return res.status(400).json({
+        error: 'Missing collection type for the request',
+        msg: 'Ensure the key is defined as \'collection\'',
+      });
+    }
+    let verifyColl;
+    try {
+      verifyColl = await mongoose.model(typeCollection).findById(id).exec();
```

Comment:
Code duplication, and an extra, unnecessary query to the database. The called method below performs this check and all necessary data validations, as stated above.

_This also goes for everywhere else you did same in this file. . ._